djstripe/models/webhooks.py

```diff
@@ -29,26 +33,78 @@ class WebhookEndpoint(StripeModel):
         blank=True,
         help_text="The API version events are rendered as for this webhook endpoint.",
     )
+    application = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="The ID of the associated Connect application.",
+    )
+    enabled_events = JSONField(
+        null=True,
+        blank=True,
+        help_text=(
+            "The list of events to enable for this endpoint. [’*’] indicates that all events"
+            " are enabled, except those that require explicit selection."
+        ),
+    )
+    # ! secret will need to be set manually
     secret = models.CharField(
         max_length=256,
+        unique=True,
+        null=True,
         blank=True,
+        validators=[RegexValidator(regex=WEBHOOK_ENDPOINT_SECRET_REGEX)],
         help_text="The endpoint's secret, used to generate webhook signatures.",
     )
     status = StripeEnumField(
         enum=WebhookEndpointStatus,
         help_text="The status of the webhook. It can be enabled or disabled.",
     )
-    url = models.URLField(help_text="The URL of the webhook endpoint.", max_length=2048)
-    application = models.CharField(
-        max_length=255,
-        blank=True,
-        help_text="The ID of the associated Connect application.",
+    url = models.URLField(
+        help_text="The URL of the webhook endpoint.",
+        max_length=2048,
     )
 
     djstripe_uuid = models.UUIDField(
         default=uuid4,
-        help_text="A UUID specific to dj-stripe generated for the endpoint",
+        unique=True,
+        help_text="A UUID specific to dj-stripe generated for the endpoint used to match the webhook url with Stripe Account using the provided keys",
     )
+    # stripe has no check for tolerance being Integral or non-negative
+    djstripe_tolerance_ms = models.FloatField(
```

Comment:
I will not be merging in this and the validation type as WE fields; it would complicate the transition. It's something we can go back on in the future.